.luacheckrc

```diff
@@ -0,0 +1,210 @@
+files["package/**/check_site.lua"] = {
+	read_globals = {
+		"alternatives",
+		"extend",
+		"in_domain",
+		"in_site",
+		"need",
+		"need_alphanumeric_key",
+		"need_array",
+		"need_array_of",
+		"need_boolean",
+		"need_chanlist",
+		"need_domain_name",
+		"need_number",
+		"need_one_of",
+		"need_string",
+		"need_string_array",
+		"need_string_array_match",
+		"need_string_match",
+		"need_table",
+		"need_value",
+		"obsolete",
+		"table_keys",
+		"this_domain",
+	},
+}
+
+files["package/**/files/lib/gluon/ebtables/*"] = {
+	read_globals = {
+		"chain",
+		"rule",
+	},
+	max_line_length = false,
+}
+
+files["package/**/luasrc/lib/gluon/config-mode/*"] = {
+	globals = {
+		"DynamicList",
+		"Flag",
+		"Form",
+		"i18n",
+		"ListValue",
+		"renderer.render",
+		"renderer.render_string",
+		"Section",
+		"TextValue",
+		"_translate",
+		"translate",
+		"translatef",
+		"Value",
+	},
+}
+
+files["package/**/luasrc/lib/gluon/**/controller/*"] = {
+	read_globals = {
+		"_",
+		"alias",
+		"call",
+		"entry",
+		"model",
+		"node",
+		"template",
+	},
+}
+
+files["scripts/*.lua"] = {
+--	allow_defined = true,
+--	module = true,
+	read_globals = {
+		-- global functions
+		"check_devices",
+		"config",
+		"config_message",
+		"config_package",
+		"devices",
+		"escape",
+		"exec",
+		"exec_capture",
+		"exec_capture_raw",
+		"exec_raw",
+		"packages",
+		"try_config",
+		-- global vars
+		"env",
+		"envtrue",
+		"images",
+		"opkg",
+		"site_code",
+		"target_packages",
+	},
+}
+
+
+files["scripts/check_site.lua"] = {
+--	allow_defined = true,
+--	module = true,
+	read_globals = {
+		"alternatives",
+		"extend",
+		"in_domain",
+		"in_site",
+		"need",
+		"need_alphanumeric_key",
+		"need_array",
+		"need_array_of",
+		"need_boolean",
+		"need_chanlist",
+		"need_domain_name",
+		"need_number",
+		"need_one_of",
+		"need_string",
+		"need_string_array",
+		"need_string_array_match",
+		"need_string_match",
+		"need_table",
+		"need_value",
+		"obsolete",
+		"table_keys",
+		"this_domain",
+	},
+}
+
+files["scripts/common.inc.lua"] = {
+--	allow_defined = true,
+--	module = true,
+	globals = {
+		-- global functions
+		"check_devices",
+		"config",
+		"defaults",
+		"device",
+		"env",
+		"envtrue",
+		"escape",
+		"exec",
+		"exec_capture",
+		"exec_capture_raw",
+		"exec_raw",
+		"factory_image",
+		"no_opkg",
+		"packages",
+		"sysupgrade_image",
+		"try_config",
+		-- global vars
+		"devices",
+		"env",
+		"envtrue",
+		"images",
+		"opkg",
+		"site_code",
+		"target_packages",
+	},
+}
+
+files["scripts/target_config.lua"] = {
+--	allow_defined = true,
+--	module = true,
+	globals = {
+		"config",
+		"config_message",
+		"config_package",
+	},
+}
+
+files["scripts/target_config_check.lua"] = {
+--	allow_defined = true,
+--	module = true,
+	globals = {
+		"config",
+		"config_message",
+		"config_package",
+	},
+}
+
+files["targets/*"] = {
+	read_globals = {
+		"config",
+		"defaults",
+		"device",
+		"env",
+		"envtrue",
+		"exec_capture_raw",
```

Comment:
Let's also add the other 3 exec variants from common.inc.lua, even if they aren't used in targets at the moment.

Reply:
And the new helper `include`.

Reply:
Added them